domain/src/main/java/no/difi/meldingsutveksling/receipt/ConversationService.java

```diff
@@ -121,21 +121,16 @@ public Conversation registerConversation(EduDocument eduDocument) {
         return repo.save(c);
     }
 
-    public void setServiceIdentifier(String conversationId, ServiceIdentifier si) {
+    public void changeServiceIdentifier(String conversationId, ServiceIdentifier si) {
         Optional<Conversation> find = repo.findByConversationId(conversationId).stream().findFirst();
         if (find.isPresent()) {
             Conversation c = find.get();
             c.setServiceIdentifier(si);
+            if (POLLABLES.contains(si)) {
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') This call to "contains()" may be a performance hot spot if the collection is large. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonar-app01.difi.local:9000/coding_rules#rule_key=squid%3AS2250)